dvc/cache/base.py

```diff
@@ -138,7 +138,7 @@ def changed(self, path_info, hash_info, filter_info=None):
             "checking if '%s'('%s') has changed.", path_info, hash_info
         )
 
-        if not self.tree.exists(path):
```

Comment:
One of the sources of dvcignore-related problems, as `self.tree`(cache tree) doesn't use it.